### PR TITLE
Added exponential backoff and retries on stackdriver api

### PR DESCRIPTION
**Pull request by olvesh**
As per discussion with Google Cloud Support this is their preferred
workaround on 503 errors from their APIs.

Also using a timeout of 10 seconds for the client to make sure we do not
have hanging requests.

I think the default values are sane enough to not adding them as program arguments.

### CODE REVIEW

File: stackdriver_exporter.go [hunk 1]
```diff
@@ -6,15 +6,15 @@ import (
 	"os"
 	"strings"
 
+	"github.com/PuerkitoBio/rehttp"
+	"github.com/frodenas/stackdriver_exporter/collectors"
```
Comment (bboreham) [inline stackdriver_exporter.go]: Minor point, but this line has moved unnecessarily.

I think the pattern is to split imports into groups: stdlib, other projects, other packages in this project.

**Comment by dmbi** [comment]
Any updates on this PR? The exponential backoff would really come in handy.

**Comment by olvesh** [comment]
It’s on me. Forgot about the feedback above. I’ll try to get time tomorrow.

**Comment by dmbi** [comment]
Thanks @olvesh !

**Comment by olvesh** [comment]
Hopefully this can be merged now.. Fixed the ordering of the imports.

**Pull request merged**